aten/src/ATen/NamedTensorUtils.cpp

```diff
@@ -470,6 +470,26 @@ optional<std::vector<Dimname>> compute_matmul_outnames(
   return compute_matmul_outnames(self.names(), other.names());
 }
 
+optional<std::vector<Dimname>> compute_cdist_outnames(
+    const Tensor& self,
+    const Tensor& other) {
+  if (!self.has_names() && !other.has_names()) {
+    return nullopt;
+  }
+  const auto self_names = self.names();
+  const auto other_names = other.names();
+
+  const auto self_batch = TensorNames(self_names, 0, num_batch_dims(self_names));
+  const auto other_batch = TensorNames(other_names, 0, num_batch_dims(other_names));
+
+  auto result = self_batch.unifyFromRight(other_batch, "cdist");
+  result.append(TensorName(self_names, -2));
+  result.append(TensorName(other_names, -2));
```

Comment:
Can you, please, leave a in-place comment that will explain '-2' value?

Reply:
Will do